k2/addons/k2-serial/zserial/serial.py

```diff
@@ -123,27 +127,36 @@ def suspended(self):
             self.resume()
 
     def suspend(self):
-        logger.debug('Suspending serial port for entity {entity}'.format(entity=self._sut.entity))
+        logger.debug('Suspending serial port for entity {entity}'.format(entity=self._entity))
         return self._messagebus.send_request(
-            SERIAL_SUSPEND, SERIAL_ENDPOINT, entity=self._sut.entity).wait()[0].result()
+            SERIAL_SUSPEND, SERIAL_ENDPOINT, entity=self._entity).wait()[0].result()
 
     def resume(self):
-        logger.debug('Resuming serial port for entity {entity}'.format(entity=self._sut.entity))
+        logger.debug('Resuming serial port for entity {entity}'.format(entity=self._entity))
         self._messagebus.send_request(
-            SERIAL_RESUME, SERIAL_ENDPOINT, entity=self._sut.entity).wait()[0].result()
+            SERIAL_RESUME, SERIAL_ENDPOINT, entity=self._entity).wait()[0].result()
 
 
 @CommandExtension(
     name='zserial',
     extends=[RUN_COMMAND],
     config_options=[
-        ConfigOption(SUT, required=True, instantiate_on=True),
+        ConfigOption(SERIAL_PORT_IDS, required=True, instantiate_on=True),
         ConfigOption(SERIAL_ENABLED, required=True),
         ConfigOption(SERIAL_DEVICE, required=False),
         ConfigOption(SERIAL_BAUDRATE, required=False),
         ConfigOption(SERIAL_PROMPT, required=False),
         ConfigOption(SERIAL_FILTERS, required=False),
         ConfigOption(SERIAL_TIMEOUT, required=False),
+        ConfigOption(SUT, required=False),
+        ConfigOption(SUT_SERIAL_PORTS, required=False),
+        ConfigOption(SUT_SERIAL_BAUDRATE, required=False),
+        ConfigOption(SUT_SERIAL_DEVICE, required=False),
+        ConfigOption(SUT_SERIAL_ENABLED, required=False),
+        ConfigOption(SUT_SERIAL_FILTERS, required=False),
+        ConfigOption(SUT_SERIAL_LOG_ENABLED, required=False),
+        ConfigOption(SUT_SERIAL_PROMPT, required=False),
+        ConfigOption(SUT_SERIAL_TIMEOUT, required=False),
```

Comment:
Even if many of these options are not used by this extension, I had problems passing the systests without them listed here. I'm guessing it's because this extension gets disabled before the `SerialLogSourceExtension` gets a chance to add the necessary options, but I didn't investigate further

Reply:
Nope, it's not the load order. `SerialLogSourceExtension` has load order 91 and `SerialExtension` has load order 1000 (default for `@CommandExtension`)